netengine/backends/snmp/openwrt.py

```diff
@@ -349,60 +339,206 @@ def interfaces_to_dict(self):
-                    'tx_bytes': tx_bytes,
-                    'state': state,
-                    'mtu': mtu,
-                    'speed': speed,
+                    'statistics': {
+                        "mac": mac_address,
+                        "type": if_type,
+                        "up": up,
+                        'rx_bytes': rx_bytes,
+                        'tx_bytes': tx_bytes,
+                        "mtu": mtu,
+                    },
                 }
             )
             results.append(result)
         return results
 
+    @property
+    def local_time(self):
+        """
+        returns the local time of the host device as a timestamp
+        """
+        octetstr = bytes(self.get('1.3.6.1.2.1.25.1.2.0')[3][0][1])
+        size = len(octetstr)
+        # string may or may not contain timezone, so size can be 8 or 11
+        if size == 8:
+            (year, month, day, hour, minutes, seconds, deci_seconds,) = struct.unpack(
+                '>HBBBBBB', octetstr
+            )
+            return int(
+                datetime.datetime(
+                    year,
+                    month,
+                    day,
+                    hour,
+                    minutes,
+                    seconds,
+                    deci_seconds * 100_000,
+                    tzinfo=pytz.utc,
+                ).timestamp()
+            )
+        elif size == 11:
+            (
+                year,
+                month,
+                day,
+                hour,
+                minutes,
+                seconds,
+                deci_seconds,
+                direction,
+                hours_from_utc,
+                minutes_from_utc,
+            ) = struct.unpack('>HBBBBBBcBB', octetstr)
+            offset = datetime.timedelta(hours=hours_from_utc, minutes=minutes_from_utc)
+            if direction == b'-':
+                offset = -offset
+            return int(
+                datetime.datetime(
+                    year,
+                    month,
+                    day,
+                    hour,
+                    minutes,
+                    seconds,
+                    deci_seconds * 100_000,
+                    tzinfo=pytz.utc,
+                ).timestamp()
+            )
+
     @property
     def RAM_total(self):
         """
         returns the total RAM of the device
         """
         return int(self.get_value('1.3.6.1.2.1.25.2.3.1.5.1'))
 
+    @property
+    def RAM_shared(self):
+        """
+        returns the shared RAM of the device
+        """
+        return int(self.get_value('1.3.6.1.2.1.25.2.3.1.6.8'))
+
+    @property
+    def RAM_cached(self):
+        """
+        returns the cached RAM of the device
+        """
+        return int(self.get_value('1.3.6.1.2.1.25.2.3.1.5.7'))
+
+    @property
+    def RAM_free(self):
+        """
+        returns the free RAM of the device
+        """
+        RAM_used = int(self.get_value('1.3.6.1.2.1.25.2.3.1.6.1'))
```

Comment:
If we are fetching `used` memory then why we are not showing it?

Reply:
Only because the [devicemonitoring](https://netjson.org/rfc.html#name-devicemonitoring) spec didn't mention it. If we use it somewhere then yes we can show it :+1: